### PR TITLE
Feat: Add liquidate and liquidateQuick methods

### PR DESCRIPTION
**Pull request by aki-0517**
## Overview
This PR adds built-in liquidation functionality to the Scallop SDK. 

### Changes
- Added `liquidate()` method: Low-level liquidation function for direct contract interaction
- Added `liquidateQuick()` method: High-level convenience function that automatically:
  - Updates oracle prices for debt and collateral assets
  - Selects and prepares coins for liquidation
  - Handles remaining coins and object references
  - Converts obligation IDs to SharedObjectRef format
- Added TypeScript type definitions for both methods in `CoreNormalMethods` and `CoreQuickMethods`

### Testing
- Added comprehensive test cases covering both `liquidate()` and `liquidateQuick()` methods
  - Tests validate:
    - Transaction construction and inspection
    - Coin selection and remaining coin handling
    - SharedObjectRef conversion for obligation objects
    - Edge case handling (missing obligations)
- Checked compatibility with existing liquidator

### CODE REVIEW

File: src/types/builder/core.ts [hunk 1]
```diff
@@ -81,6 +81,12 @@ export type CoreNormalMethods = {
     loan: SuiObjectArg,
     poolCoinName: string
   ) => void;
+  liquidate: (
+    obligation: SuiObjectArg,
+    coin: SuiObjectArg,
+    debtCoinName: string,
+    collateralCoinName: string
+  ) => TransactionResult;
```
[the remark this answers is not in the document]
Comment (aki-0517) [reply src/types/builder/core.ts]: Like existing methods (such as borrowFlashLoan), the type definition is aligned with TransactionResult, matching the implementation's style of handling multiple return values. Since this aligns with the current codebase, it remains unchanged.

File: test/builder.spec.ts [hunk 2]
```diff
@@ -334,6 +334,110 @@ describe('Test Scallop Core Builder', () => {
...
+    try {
+      const result = await tx.liquidateQuick(
+        liquidationAmount,
+        debtCoinName,
+        collateralCoinName,
+        obligationId,
+        '1'
+      );
+
+      tx.transferObjects([result], sender);
```
[the remark this answers is not in the document]
Comment (aki-0517) [reply test/builder.spec.ts]: It should be received as a tuple, as pointed out. I've fixed the test to transfer both extraDebtCoin and collateralCoin.

File: src/types/builder/core.ts [hunk 3]
```diff
@@ -152,6 +158,19 @@ export type CoreQuickMethods = {
...
+    collateralCoinName: string,
+    obligationId: SuiObjectArg,
+    obligationInitialSharedVersion?: string,
+    updateOracleOptions?: {
+      usePythPullModel?: boolean;
+      useOnChainXOracleList?: boolean;
+      sponsoredFeeds?: string[];
+      isSponsoredTx?: boolean;
+    }
+  ) => Promise<TransactionResult>;
```
[the remark this answers is not in the document]
Comment (aki-0517) [reply src/types/builder/core.ts]: Like `liquidate`, it maintains a single `TransactionResult` type to align with existing patterns. Since execution results can return multiple values, this structure prioritizes compatibility with existing APIs.

File: src/builders/coreBuilder.ts [hunk 4]
```diff
@@ -489,6 +508,56 @@ const generateCoreQuickMethod: GenerateCoreQuickMethod = ({
         updateOracleOptions
       );
     },
+    liquidateQuick: async (
+      amount,
+      debtCoinName,
+      collateralCoinName,
+      obligationId,
+      obligationInitialSharedVersion = '1',
```
[the remark this answers is not in the document]
Comment (aki-0517) [reply src/builders/coreBuilder.ts]: The existing Quick API also has a default value, so retains `= ‘1’` to prioritize backward compatibility.

**Comment by fum-is-chum** [comment]
Some notes:
- The test will fail because we cannot liquidate healthy obligation
- We do not need to manage the `initialSharedVersion` of the obligation object, we can let the sdk to determine the version on the transaction building process
- In order to liquidate the obligation, we need to update all collaterals and debts of the obligation, not just the liquidated collateral and repaid debt
- Liquidation method returns exactly two nested results

Other than that all good!

**Review by fum-is-chum — APPROVED** [review]
